customers/services/profile.py

```diff
@@ -127,13 +133,117 @@ def get_profile(self, id: UUID) -> HelsinkiProfileUser:
+        are replaced by empty values.
+
+        force_only_one: bool -> If more than one profile is found, it will raise an error
+        """
+        query = f"""
+            query FindProfile {{
+                profiles(
+                    serviceType: BERTH,
+                    firstName: "{first_name}",
+                    lastName: "{last_name}",
+                    emails_Email: "{email}",
+                    phones_Phone: "{phone}"
+                ) {{
+                    edges {{
+                        node {{
+                            id
+                            first_name: firstName
+                            last_name: lastName
+                            primary_email: primaryEmail {{
+                                email
+                            }}
+                            primary_phone: primaryPhone {{
+                                phone
+                            }}
+                        }}
+                    }}
+                }}
+            }}
+        """
+        response = self.query(query)
+        profiles = (
+            response.get("profiles", {}).get("edges", [])
+            if response
+            and response.get("profiles", {})
+            and response.get("profiles", {}).get("edges")
+            else []
+        )
+
+        if force_only_one:
+            if len(profiles) > 1:
+                ids = [from_global_id(node.get("id")) for node in profiles]
+                raise MultipleProfilesException(ids=ids)
+            elif len(profiles) == 0:
+                raise NoProfilesException
+
+        users = []
+        for profile_edge in profiles:
+            users.append(self.parse_user_edge(profile_edge))
+
+        if force_only_one:
+            return users[0]
+
+        return users
+
+    @transaction.atomic
+    def create_profile(
+        self, first_name: str, last_name: str, email: str = None, phone: str = None
+    ):
+        from ..models import CustomerProfile
+
+        variables = {
+            "input": {"profile": {"firstName": first_name, "lastName": last_name}}
+        }
+        if email:
+            variables["input"]["profile"]["addEmails"] = [
+                {"primary": True, "email": email, "emailType": "NONE"}
+            ]
+        if phone:
+            variables["input"]["profile"]["addPhones"] = [
+                {"primary": True, "phone": phone, "phoneType": "NONE"}
+            ]
+
+        mutation = """
+        mutation CreateProfile($input: CreateProfileMutationInput!) {
+            createProfile(input: $input) {
+                profile {
+                    id
+                }
+            }
+        }
+        """
+        response = self.query(query=mutation, variables=variables)
+        profile = None
+        if global_id := response.get("createProfile", {}).get("profile", {}).get("id"):
+            profile_id = from_global_id(global_id)
+            profile = CustomerProfile.objects.create(id=profile_id)
```

Comment:
Should there be an else: branch that logs some error info from response if global_id is not found?

Reply:
Hmmm. I _think_ the only case where the global id would be `None` is if there's some error on the Profile service, and that's being caught now by the `query` method, so probably we don't even need this condition.

```python
r = requests.post(url=self.api_url, json=body, headers=headers)
r.raise_for_status()
response = r.json()
if errors := response.get("errors"):
    raise ProfileServiceException(str(errors))
```

Reply:
ok